pkg/scheduler/preemption/preemption_test.go

```diff
@@ -43,26 +43,34 @@ import (
 	"sigs.k8s.io/kueue/pkg/constants"
 	controllerconstants "sigs.k8s.io/kueue/pkg/controller/constants"
 	"sigs.k8s.io/kueue/pkg/features"
-	"sigs.k8s.io/kueue/pkg/hierarchy"
 	"sigs.k8s.io/kueue/pkg/scheduler/flavorassigner"
 	"sigs.k8s.io/kueue/pkg/util/slices"
 	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-var snapCmpOpts = []cmp.Option{
-	cmpopts.EquateEmpty(),
-	cmpopts.IgnoreUnexported(hierarchy.Cohort[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{}),
-	cmpopts.IgnoreUnexported(hierarchy.ClusterQueue[*cache.CohortSnapshot]{}),
-	cmpopts.IgnoreUnexported(hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]{}),
-	cmpopts.IgnoreFields(cache.ClusterQueueSnapshot{}, "AllocatableResourceGeneration"),
-	cmp.Transformer("Cohort.Members", func(s sets.Set[*cache.ClusterQueueSnapshot]) sets.Set[kueue.ClusterQueueReference] {
-		result := make(sets.Set[kueue.ClusterQueueReference], len(s))
-		for cq := range s {
-			result.Insert(cq.Name)
+type nodeKey struct {
+	cohort bool
+	name   string
+}
+
+func resourceNodes(snapshot *cache.Snapshot) map[nodeKey]cache.ResourceNode {
+	nodes := map[nodeKey]cache.ResourceNode{}
```

Comment:
nit: prealloc for len(Cohorts) + len(CQs)

Reply:
Done.

However: is this much utility? I ran pprof on a test suite recently, and the overwhelming majority of CPU is on the test fixture, or some serialization related to k8 client/object. I would argue in general keeping test definitions as short as possible, which would also include not preallocating maps/slices/etc

edit: Just ran pprof now, repeating the test suite 100 times. Ironically the optimized version spent 20% more time in this function (but both numbers were really small, 100ms vs 120ms, out of ~35 seconds of samples). Time spent doing GC also had negligible difference

Reply:
Let's leave it as is now, to not do another round-trip on this PR. If we decide to clean this up, we can do it throughout entire codebase in 1 PR